perf_test/sparse/KokkosSparse_pcg.cpp

```diff
@@ -307,27 +307,49 @@ int main (int argc, char ** argv){
   }
 
 
+  // Prepare for Kokkos::initialize() call
+  Kokkos::InitArguments init_args; // Construct with default args, change members based on exec space
 
```

Comment:
all this num_threads etc. setting should happen independent of the backend. You can always parse all the command line options and set those things. Even for CUDA. No need for backend specific code paths here.

Reply:
This test is a bit funky, since it is likely run by a script it may be ready to run with all three backends enabled since `exec_space::initialize()` and `exec_space::finalize()` were called within backend-specific if-guarded code. I think what I did would allow one backend's num_threads to overwrite the other's in this case. I'm going to change this so initialize and finalize are called within each backend's if-guards with the num_threads etc. parsing happening there.